src/main/java/org/rumbledb/compiler/StaticContextVisitor.java

```diff
@@ -179,20 +179,27 @@ public StaticContext visitLetClause(LetClause clause, StaticContext argument) {
     }
 
     @Override
-    public StaticContext visitGroupByClauseVar(GroupByClauseVar expression, StaticContext argument) {
-        StaticContext groupByClauseContext;
-        if (expression.getExpression() != null) {
-            // if a variable declaration takes place
-            this.visit(expression.getExpression(), argument);
-            // initialize execution and storage modes and then add the variable to the context
-            expression.initHighestExecutionAndVariableHighestStorageModes(this.visitorConfig);
-            groupByClauseContext = visitFlowrVarDeclaration(expression, argument);
-        } else {
-            // if a variable is only referenced, use the context as is
-            groupByClauseContext = argument;
+    public StaticContext visitGroupByClause(GroupByClause clause, StaticContext argument) {
+        StaticContext groupByClauseContext = new StaticContext(argument);
+        for (GroupByVariableDeclaration variable : clause.getGroupVariables()) {
+            if (variable.getExpression() != null) {
+                // if a variable declaration takes place
+                this.visit(variable.getExpression(), argument);
+                // initialize execution and storage modes and then add the variable to the context
```

Comment:
I think this comment is outdated/redundant now.

Reply:
I removed it in the other branch